libraries/WiFi/src/WiFiMulti.h

```diff
@@ -42,10 +43,31 @@ class WiFiMulti
 
     bool addAP(const char* ssid, const char *passphrase = NULL);
 
+    void setStrictMode(bool bStrict);
+    void setAllowOpenAP(bool bAllowOpenAP);
+    void setTestConnection(bool bTestConnection);
+    void setTestPhrase(const char* testPhrase);
+    void setTestURL(String url);
+
     uint8_t run(uint32_t connectTimeout=5000);
 
+    void APlistClean(void);
+
 private:
     std::vector<WifiAPlist_t> APlist;
+
+    bool _bStrict = true;
+    bool _bAllowOpenAP = false;
+    bool _bTestConnection = false;
+    String _testPhrase = "301 Moved";
+    String _testURL = "http://www.amazon.com";
```

Comment:
Idk if you need some specific kind of page but google's URL would work for most places, including China.